mongo/src/main/java/com/redhat/lightblue/mongo/crud/BsonMerge.java

```diff
@@ -175,8 +175,13 @@ public boolean merge(DBObject oldDoc,DBObject newDoc) {
                     if(hidden) {
                         // Is the removed field in metadata?
                         try {
-                            md.resolve(removedField);
-                            hidden=false;
+                            // don't try to resolve a hidden field
+                            if (!removedField.equals(Translator.HIDDEN_SUB_PATH)) {
```

Comment:
removedField is a path, you should check if HIDDEN_SUB_PATH is one of its segments, equals won't work.